src/cmd/fix.rs

```diff
@@ -388,11 +504,51 @@ async fn check_integrations<L: Logger>(logger: &L, config: &Config) -> Vec<Error
 }
 
 #[tracing::instrument]
-async fn check_integration(integration: &Integration) -> Result<(), Error> {
+async fn check_integration_connection(integration: &Integration) -> Result<(), Error> {
     let Protocol::Git(transport) = integration.protocol();
-    repository::ls_remote(transport)
+    repository::ls_remote(transport).await.or_else(|err| {
+        Error::integration_connection_error(integration.remote(), transport, err).wrap_err()
+    })?;
+    Ok(())
+}
+
+#[tracing::instrument]
+async fn check_integration_scan(
+    ctx: &AppContext,
+    config: &Config,
+    integration: &Integration,
+) -> Result<(), Error> {
+    let remote = integration.remote();
+    let cli = fossa_cli::find_or_download(ctx, config.debug().location(), DesiredVersion::Latest)
         .await
-        .or_else(|err| Error::integration_error(integration.remote(), transport, err).wrap_err())?;
+        .or_else(|_err| Error::download_cli_error(remote).wrap_err())?;
+
+    let references = integration.references().await.unwrap_or_default();
+    if references.is_empty() {
+        return Error::EmptyReferences.wrap_err();
+    }
+    let mut reference = &references.clone()[0];
+
+    let main_reference = references
+        .into_iter()
+        .filter(|r| r.get_name().as_str() == MAIN_BRANCH || r.get_name().as_str() == MASTER_BRANCH)
+        .collect::<Vec<_>>();
+
+    if !main_reference.is_empty() {
+        reference = &main_reference[0];
+    }
+
+    let cloned_location = integration
+        .clone_reference(reference)
+        .await
+        .or_else(|_err| Error::CloneReference.wrap_err())?;
+
+    let job = ScanGitVCSReference::new(integration, reference);
+
+    cli.analyze(job.get_scan_id(), cloned_location.path())
+        .await
+        .or_else(|_err| Error::integration_scan_error(remote, reference.get_name()).wrap_err())?;
```

Comment:
The message was too big and overloaded with unnecessary debug details. In our previous conversation you said that I could parse the error message for only the fossa analyze error output or I could just not display the error and point users to running the fossa analyze through the broker fix explanation. I chose to do the latter so that is why I'm discarding the error.

Reply:
Makes sense! Whenever we do surprising things (like throw away errors) we should comment why we're doing so.